modules/apps/portlet-configuration/portlet-configuration-web/src/main/resources/META-INF/resources/edit_permissions.jsp

```diff
@@ -132,24 +131,70 @@ if (Validator.isNotNull(portletConfigurationPermissionsDisplayContext.getModelRe
 							}
 
-							dataMessage = HtmlUtil.escapeAttribute(LanguageUtil.format(request, preselectedMsg, new Object[] {role.getTitle(locale), _getActionLabel(request, resource.getName(), action), type, HtmlUtil.escape(portletConfigurationPermissionsDisplayContext.getGroupDescriptiveName())}, false));
+							dataMessage = HtmlUtil.escapeAttribute(LanguageUtil.format(request, preselectedMsg, new Object[] {role.getTitle(locale), _getActionLabel(request, resourceName, action), type, HtmlUtil.escape(portletConfigurationPermissionsDisplayContext.getGroupDescriptiveName())}, false));
+
+							if (GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-87806"))) {
+								disabled = true;
+							}
 						}
 
 						String actionSeparator = Validator.isNotNull(preselectedMsg) ? ActionUtil.PRESELECTED : ActionUtil.ACTION;
 					%>
 
 						<liferay-ui:search-container-column-text
 							cssClass="table-column-text-center"
-							name="<%= _getActionLabel(request, resource.getName(), action) %>"
+							name="<%= _getActionLabel(request, resourceName, action) %>"
 						>
 							<c:if test="<%= disabled && checked %>">
 								<input name="<%= liferayPortletResponse.getNamespace() + role.getRoleId() + actionSeparator + action %>" type="hidden" value="<%= true %>" />
 							</c:if>
 
-							<input <%= checked ? "checked" : StringPool.BLANK %> class="<%= Validator.isNotNull(preselectedMsg) ? "lfr-checkbox-preselected lfr-portal-tooltip" : StringPool.BLANK %>" title="<%= dataMessage %>" <%= disabled ? "disabled" : StringPool.BLANK %> id="<%= FriendlyURLNormalizerUtil.normalize(role.getName()) + actionSeparator + action %>" name="<%= liferayPortletResponse.getNamespace() + role.getRoleId() + actionSeparator + action %>" onclick="<%= Validator.isNotNull(preselectedMsg) ? "return false;" : StringPool.BLANK %>" type="checkbox" />
+							<c:choose>
+								<c:when test='<%= GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-87806")) %>'>
+									<div>
+										<div class="custom-checkbox custom-control custom-control-inline">
+											<label>
+												<input
+													<%= (checked || indeterminate) ? "checked" : StringPool.BLANK %>
+													<%= disabled ? "disabled" : StringPool.BLANK %>
+													<%= indeterminate ? "value=\"indeterminate\"" : StringPool.BLANK %>
+													class="custom-control-input"
+													name="<%= liferayPortletResponse.getNamespace() + role.getRoleId() + actionSeparator + action %>"
+													type="checkbox"
+												/><span class="custom-control-label"></span
+											>
+											</label>
+										</div>
+
+										<react:component
+											module="js/PermissionsCheckbox"
+											props='<%=
+												HashMapBuilder.<String, Object>put(
+													"checked", checked
+												).put(
+													"disabled", disabled
+												).put(
+													"indeterminate", indeterminate
+												).put(
+													"indeterminateValue", "indeterminate"
```

Comment:
Maybe we could set this "indeterminateValue" in the component as default value.... and remove it from here?